### ISSUE TITLE
Zellij does not exit on reboot/shutdown

### ISSUE THREAD

**Issue by langbamit** [zellij-org/zellij#1414]
**Current Behavior**
Zellij doesn't quit. 
So I get the nice message of Systemd waiting for a stop job, for 90 seconds. Until zellij gets killed. In the log it looks like that:
```
May 12 14:20:11 systemd[1]: session-3.scope: Failed with result 'timeout'.
May 12 14:20:11 systemd[1]: session-3.scope: Killing process 3103 (input_handler) with signal SIGKILL.
May 12 14:20:11 systemd[1]: session-3.scope: Killing process 3002 (zellij) with signal SIGKILL.
May 12 14:20:11 systemd[1]: session-3.scope: Stopping timed out. Killing.
May 12 14:18:42 systemd[1]: Finished Save Hardware Clock.
May 12 14:18:42 systemd[1]: save-hwclock.service: Deactivated successfully.
May 12 14:18:42 systemd[1]: Stopped Network Manager Script Dispatcher Service.
```
If I quit zellij before reboot/shutdown, everything is fine (very fast on reboot/shutdown).

**Expected Behavior**
When rebooting or shuting down the computer, zellij should quit properly.

**Basic information**

`zellij --version`: zellij 0.29.1
`uname -av` or `ver`(Windows): Linux thinkpad 5.17.5 #1-NixOS SMP PREEMPT Wed Apr 27 12:41:17 UTC 2022 x86_64 GNU/Linux
`alacritty --version`: alacritty 0.10.1 ()

**Comment by raphCode** [comment]
I can see similar messages appearing in my journal of my server in the past.
I tested a bit around and was able to pinpoint the issue to a stuck zellij client process.
This is already tracked here: #1326, but that it also impacts local sessions is new to me. I'll update the referenced issue.

**Comment by AjiBuster499** [comment]
Hi, I encountered this issue recently, and tracked it down to applications open inside Zellij as well (i.e. leaving a Neovim instance open). I'm on v0.30, so the PR that closed this should be in there. I don't know if something changed or what.

**Comment by raphCode** [comment]
Hey thanks for the report, @AjiBuster499!
Can you check if you got the same messages in `journalctl` as in the first post?

**Comment by AjiBuster499** [comment]
> Did you got the same messages in `journalctl` as in the first post?

Yes, I just rebooted my machine to check, and I got the same messages.

```
Jun 21 11:05:19 arch audit: BPF prog-id=0 op=UNLOAD
Jun 21 11:05:19 arch audit: BPF prog-id=0 op=UNLOAD
Jun 21 11:06:49 arch systemd[1]: session-3.scope: Stopping timed out. Killing.
Jun 21 11:06:49 arch systemd[1]: session-3.scope: Killing process 9894 (zellij) with signal SIGKILL.
Jun 21 11:06:49 arch systemd[1]: session-3.scope: Killing process 9901 (input_handler) with signal SIGKILL.
Jun 21 11:06:49 arch systemd[1]: session-3.scope: Failed with result 'timeout'.
```
On Arch Linux 5.18.5-arch1-1

**Comment by raphCode** [comment]
Oh well...

Playing around, this seems to happen when the server is killed first (SIGTERM), then the client hangs and does not respond to SIGTERM anymore.
We fixed the opposite situation in #1454, but the actual communication problem is symmetric so the client needs a fix as well.

**Comment by tlinford** [comment]
> Oh well...
> 
> Playing around, this seems to happen when the server is killed first (SIGTERM), then the client hangs and does not respond to SIGTERM anymore. We fixed the opposite situation in #1454, but the actual communication problem is symmetric so the client needs a fix as well

I found that a couple of days ago too, have a pr upcoming.

**Comment by raphCode** [comment]
i just comitted something as well :D

**Comment by tlinford** [comment]
nice :)
